pkg/track/events.go

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+package track
+
+import (
+	"context"
+	"sync"
+
+	v1 "github.com/cilium/cilium/pkg/hubble/api/v1"
+	"github.com/microsoft/retina/pkg/enricher"
+	"github.com/microsoft/retina/pkg/log"
+)
+
+type Track struct {
+	ctx             context.Context
```

Comment:
`context.Context` shouldn't be stored in a struct: see: https://pkg.go.dev/context

> Do not store Contexts inside a struct type; instead, pass a Context explicitly to each function that needs it.

Reply:
Yes, `Start` take a context and lifecycle is tied to it. This variable is not actually needed, what I do need it the `cancelFunc` (in lieu of a separate done channel) to make sure I stop the track before I close the external channel.